lib/Tools/LinearLayout.cpp

```diff
@@ -927,9 +929,8 @@ LinearLayout LinearLayout::invertAndCompose(const LinearLayout &outer) const {
   auto BReduced = B.sublayout(BNonIdentityInDims, outDims);
 
   // If one is empty, the other must be empty as well
-  assert((AReduced == LinearLayout::empty()) ==
-         (BReduced == LinearLayout::empty()));
-  bool isEmpty = AReduced == LinearLayout::empty();
+  assert((ANonIdentityInDims.empty()) == (BNonIdentityInDims.empty()));
+  bool isEmpty = ANonIdentityInDims.empty();
```

Comment:
can a non-empty layout have empty input dims? It seems like this is something we should check that it doesn't happen when creating the LL.

Reply:
Do you suggest to prohibit creation of such layouts?

Reply:
yes, I don't think they make sense at all

Reply:
Filtering such linear layout requires rework of some algorithms, which can take some time.
I will return to this issue when clean up more urgent tasks.

Reply:
Interesting. Where do we hit these layouts?

Reply:
So far I found two places, both are using `LinearLayout::sublayout` method:
- `invertAndCompose` from this PR, which is used in `ttg.convert_layout` to llvm converter
- `GatherOpConversion::emitWarpLocalGather` in case of 1-d gather